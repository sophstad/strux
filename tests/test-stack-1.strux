int main() {
	Stack::string:: s = new Stack::string::();
	Stack::num:: s2 = new Stack::num::();
	Stack::int:: s3 = new Stack::int::();
	Stack::bool:: s4 = new Stack::bool::();
	s.add("S");
	s.pop();
	s.peek();

        s2.add(2.63);
        s2.add(5.2);
        num top = s2.peek();
        int numSize = s2.size();
        print(numSize);
        print(top);

        s3.add(35);
        int popInt = s3.peek();
        print(popInt);

        s4.add(true);
        bool topBool = s4.peek();
        print(topBool);

	return 0;
}
