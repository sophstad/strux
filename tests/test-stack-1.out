2
5.200000
35
1
